### ISSUE TITLE
The question of computation of GetErfGradient in gradient_builder.cc --02

### ISSUE THREAD

**Issue by rGitcy** [microsoft/onnxruntime#13527]
### Describe the documentation issue

#13512 
forward graph:

x->erf->output

output = erf(x)

backward graph:

gradient_of_output (dy) -> erf_grad -> gradient_of_input（dx）

The computation should be rule gradient_of_output = gradient_of_input * erf_grad(x) according to the differentiation formulas of erf.

**Comment by baijumeswani** [comment]
[rGitcy](https://github.com/rGitcy). let me try to break the code down a little more.

Let's assume that we have a loss function `L = f(w1, w2, ..., wn)`. Here w1, w2, ... wn are weights in the model.

we want to minimize the loss with respect to the weights, so we try to find the partial derivatives dL/dw1, dL/dw2...dL/dwn. These are the gradients that we want to compute (that can be used as a step in gradient descent).

Back to erf. Let's say the output of the erf node is `z` and input is `x`. So, `z = erf(x)`.

To get dL/dW we need to compute dL/dx (local gradient) which can be forwarded to the node before erf.

...-----> x -----------> (erf) ---------->z----------->....L
...<----- dL/dx<----- erf_grad<-----dL/dz<------...1 (dL/dL)

dL/dx = dL/dz * dz/dx = dL/dz * erf_grad
or gradient of input = gradient of output * f'(x)
I think the word of here should be replaced with `with respect to` to be mode precise.

dL/dx can then be used to compute dL/dw_i

Hope this made sense.

**Comment by rGitcy** [comment]
> [rGitcy](https://github.com/rGitcy). let me try to break the code down a little more.
> 
> Let's assume that we have a loss function `L = f(w1, w2, ..., wn)`. Here w1, w2, ... wn are weights in the model.
> 
> we want to minimize the loss with respect to the weights, so we try to find the partial derivatives dL/dw1, dL/dw2...dL/dwn. These are the gradients that we want to compute (that can be used as a step in gradient descent).
> 
> Back to erf. Let's say the output of the erf node is `z` and input is `x`. So, `z = erf(x)`.
> 
> To get dL/dW we need to compute dL/dx (local gradient) which can be forwarded to the node before erf.
> 
> ...-----> x -----------> (erf) ---------->z----------->....L ...<----- dL/dx<----- erf_grad<-----dL/dz<------...1 (dL/dL)
> 
> dL/dx = dL/dz * dz/dx = dL/dz * erf_grad or gradient of input = gradient of output * f'(x) I think the word `of` here should be replaced with `with respect to` to be mode precise.
> 
> dL/dx can then be used to compute dL/dw_i
> 
> Hope this made sense.

thx. i try it according to your guides.
